<?xml version="1.0"?>

<launch>

<!--  <node pkg="hector_heat_detection" type="heat_detection" name="heat_detection" output="screen">
    <remap from="thermal/image" to="/arm_thermal_cam/image_converted" /> 
  </node>-->
    
  <arg name="percept_class_id" default="heat_source" /> 
  
  <arg name="camera_nodelet_manager_name" default="arm_thermal_cam_nodelet_manager" /> 
  
  <arg name="thermal_camera_ns" default="arm_thermal_cam" /> 
  
  <node pkg="nodelet" type="nodelet"
    name="heat_detection" 
    args="load hector_image_proc/heat_detection_nodelet $(arg camera_nodelet_manager_name)" 
    output="screen" respawn="true">
    <remap from="~/thermal/image" to="/$(arg thermal_camera_ns)/image_converted"/>    
    <remap from="~/thermal/image_mapped" to="/$(arg thermal_camera_ns)/image_mapped"/>    
    <remap from="~/thermal/camera_info" to="/$(arg thermal_camera_ns)/camera_info"/>    
    <remap from="/image_percept" to="/worldmodel/image_percept" />

    <param name="min_temp_detection" value="29" />
    <param name="max_temp_detection" value="100" />
    <param name="min_area_detection" value="40" />
    <param name="min_dist_between_blobs" value="20" />
    <param name="percept_class_id" value="$(arg percept_class_id)" />
  </node>  

</launch>
